src/master/master.js

```diff
@@ -53,42 +58,24 @@ export function redactLog(log, playerID) {
 }
 
 /**
- * Verifies that the move came from a player with the
- * appropriate credentials.
+ * Verifies that the game has metadata and is using credentials.
  */
-export const isActionFromAuthenticPlayer = ({
-  action,
-  gameMetadata,
-  playerID,
-}) => {
-  if (!gameMetadata) {
-    return true;
-  }
-
-  const hasCredentials = Object.keys(gameMetadata.players).some(key => {
-    return !!(
-      gameMetadata.players[key] && gameMetadata.players[key].credentials
-    );
+export const doesGameRequireAuthentication = gameMetadata => {
+  if (!gameMetadata) return false;
+  const { players } = gameMetadata;
+  const hasCredentials = Object.keys(players).some(key => {
+    return !!(players[key] && players[key].credentials);
   });
-  if (!hasCredentials) {
-    return true;
-  }
-
-  if (!action.payload) {
-    return false;
-  }
-
-  if (!action.payload.credentials) {
-    return false;
-  }
-
-  if (
-    action.payload.credentials !== gameMetadata.players[playerID].credentials
-  ) {
-    return false;
-  }
+  return hasCredentials;
+};
 
-  return true;
+/**
+ * Verifies that the move came from a player with the correct credentials.
+ */
+export const isActionFromAuthenticPlayer = (credentials, playerMetadata) => {
```

Comment:
Let's name the first parameter `actionCredentials` so that it's a bit more descriptive?